src/propcache/_helpers_py.py

```diff
@@ -46,13 +46,13 @@ def __init__(self, wrapped: Callable[..., _T]) -> None:
         self.name = wrapped.__name__
 
     @overload
-    def __get__(self, inst: None, owner: Optional[Type[Any]] = None) -> Self: ...
+    def __get__(self, inst: None, owner: Optional[Type[object]] = None) -> Self: ...
```

Comment:
I wonder if using a `ParamSpec` would be helpful here..

Reply:
This is being used like a property decorator, so the method will always take no arguments (other than self), if I understand correctly.

So, ParamSpec seems irrelevant, but the Callable should be able to be tightened up on line 43.
So, that can probably become `Callable[[_TSelf[_T]], _T]`.

However, looking at that, it's defined the _cache as `_cache: Dict[str, _T]`. Wouldn't that make it impossible to use this decorator multiple times on different values? e.g. If I have a class with a `URL` and a port as `int`, then the dict type would need to be `dict[str, URL | int]`, which would then result in both attributes returning `URL | int`.

So, I think the _cache annotation might need some reworking...

Reply:
This can probably be addressed in a separate PR. For the scope of this PR, I think this is good to merge.